pkg/bpf/bpf_kmesh_workload.go

```diff
@@ -135,34 +127,54 @@ func (sc *BpfSockConnWorkload) close() error {
 
 func (sc *BpfSockConnWorkload) Attach() error {
 	var err error
-	cgopt := link.CgroupOptions{
+	cgopt4 := link.CgroupOptions{
 		Path:    sc.Info.Cgroup2Path,
 		Attach:  sc.Info.AttachType,
 		Program: sc.KmeshCgroupSockWorkloadObjects.CgroupConnect4Prog,
 	}
 
-	sc.Link, err = link.AttachCgroup(cgopt)
-	if err != nil {
-		return err
-	}
-
-	cgopt = link.CgroupOptions{
+	cgopt6 := link.CgroupOptions{
 		Path:    sc.Info6.Cgroup2Path,
 		Attach:  sc.Info6.AttachType,
 		Program: sc.KmeshCgroupSockWorkloadObjects.CgroupConnect6Prog,
 	}
 
-	sc.Link6, err = link.AttachCgroup(cgopt)
-
 	if GetStartType() == Restart {
-		return err
-	}
+		pinPath4 := filepath.Join(sc.Info.BpfFsPath, "sockconn_prog")
+		sclink4, err := link.LoadPinnedLink(pinPath4, &ebpf.LoadPinOptions{})
+		if err != nil {
+			return err
+		}
+		if err := sclink4.Update(cgopt4.Program); err != nil {
```

Comment:
can we abstract these logic into a helper function, like `bpfProgUpdate` , just a suggestion, may name more appropriately